python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -363,13 +363,22 @@ def indices(self):
         >>> df.groupby(by=["a"]).indices
         {10: array([0, 1]), 40: array([2])}
         """
-        group_names, offsets, _, grouped_values = self._grouped()
+        offsets, group_keys, (indices,) = self._groupby.groups(
+            [
+                cudf.core.column.as_column(
+                    range(len(self.obj)), dtype=size_type_dtype
+                )
+            ]
+        )
 
+        group_keys = libcudf.stream_compaction.drop_duplicates(group_keys)
+        if len(group_keys) > 1:
+            index = cudf.MultiIndex.from_arrays(group_keys)
+        else:
+            (group_keys,) = group_keys
```

Comment:
Is this equivalent? I think it would be clearer.
```suggestion
            group_keys = group_keys[0]
```

Reply:
I guess I already know that the length is 1, but in general I prefer singleton destructuring over indexing since it also asserts the length

Reply:
I had to open an interpreter and experiment with this snippet to ensure that it was just getting the first element and not doing something fancier. I find this hard to read, but leave it to your discretion.